CMakeLists.txt

```diff
@@ -196,6 +196,8 @@ set_target_properties(${PROJECT_NAME} PROPERTIES
     C_STANDARD_REQUIRED YES
     )
 
+target_compile_options(${PROJECT_NAME} PRIVATE $<$<BOOL:${CMAKE_COMPILER_IS_GNUCC}>:-fdata-sections -ffunction-sections>)
```

Comment:
You could use the cmake-generator-expression `$<C_COMPILER_ID:GNU>` in order to be compliant with the rest of the project.

Reply:
Done! and tested